sympy/series/limits.py

```diff
@@ -280,7 +280,7 @@ def set_signs(expr):
                 newe = e.subs(z, z + z0)
             try:
                 coeff, ex = newe.leadterm(z, cdir=cdir)
-            except ValueError:
+            except (ValueError, NotImplementedError):
```

Comment:
This is not a good fix. There are many functions that do not have leading term and series expansion methods implemented, yet. Not raising `NotImplementedError` will make it harder to debug issues caused by the missing cases.

Reply:
Yeah correct , I completely agree.  I also agree more to what you said which is 
> The natural fix is to implement the missing method instead of masking the error.

This is because as you were doubting my implementation behind the `_eval_rewrite_as_tractable` method for `lowergamma` function , I was thinking that the idea is mostly related to `_eval_rewrite_as_tractable` for the `uppergamma `function too  . This was implemented couple years ago 
```
    def _eval_rewrite_as_tractable(self, s, x, **kwargs):
        return exp(loggamma(s)) - lowergamma(s, x)
```
 Hence for some clarity I computed quite a good number of basic limits for `uppergamma ` but I realize this method might end up to be quite buggy . Even Basic limits for `upergamma` have good amount of bugs , the most basic one being 
```
>>> limit(uppergamma(a, x), a, 0)
oo
>>> limit(uppergamma(a, x), a, 0, '-')
Traceback (most recent call last):
 NotImplementedError: loggamma has no _eval_as_leading_term routine
```
The answer for this should be `expint(1, x) ` , I mean obviously if x tends to 0 here then we can expect oo but otherwise we shouldn't 
```
>>> limit(expint(1, x), x, 0)
oo
>>> limit(expint(1, x), x, 0, '-')
oo
```
Obviously if we comment this out ,we get the correct answer 
```
>>> limit(uppergamma(a, x), a, 0)
expint(1, x)
>>> limit(uppergamma(a, x), a, 0, '-')
expint(1, x)
```
![image](https://user-images.githubusercontent.com/87052487/157049407-93fa7923-6ded-48bc-8ec9-793a669e310c.png)

I didn't even open this pr considering fixing limits but now I this has pr showed me so much scope of improvements in limits

Reply:
Though fundamentally like concept wise I don't really see any obvious mistake here ..... I mean both functions used here the `exponential` one and the `lowergamma` one would end up having `aseries` , hence this doesn't seem wrong
```
    def _eval_rewrite_as_tractable(self, s, x, **kwargs):
        return exp(loggamma(s)) - lowergamma(s, x)
```